planner/core/preprocess.go

```diff
@@ -414,11 +414,14 @@ func (p *preprocessor) checkBindGrammar(originNode, hintedNode ast.StmtNode, def
 	var resNode ast.ResultSetNode
 	switch n := originNode.(type) {
 	case *ast.SelectStmt:
-		resNode = n.From.TableRefs
+		resNode = n
+	case *ast.SetOprStmt:
+		resNode = n
 	case *ast.DeleteStmt:
 		resNode = n.TableRefs.TableRefs
 	case *ast.UpdateStmt:
```

Comment:
I think we should also check where expr in Delete/Update

Reply:
Let's open a new issue about Delete/Update/Insert~

Reply:
ok~ I have created a new issue #27422

Reply:
thx a lot, Let me working on it.